src/cluster_legacy.c

```diff
@@ -2928,20 +3003,28 @@ int clusterProcessPacket(clusterLink *link) {
     clusterMsg *hdr = (clusterMsg *)link->rcvbuf;
     uint16_t type = ntohs(hdr->type);
     mstime_t now = mstime();
+    clusterNode *sender = getNodeFromLinkAndMsg(link, hdr);
```

Comment:
It's not safe to call getNodeFromLinkAndMsg for a light msg, because it can access clusterMsg fields like `hdr->sender`. This happens if the sender sends a light message as its first message on the link, before we know the node ID. It's not allowed, but we want to disconnect the node, not crash ourselves.

```suggestion
    if (!link->node || nodeInHandshake(link->node)) {
        freeClusterLink(link);
        serverLog(LL_WARNING, "Closing link for node that sent a lightweight message of type %hu as its first message on the link", type);
            return 0;
    }
    custerNode *sender = link->node;    
```

Or maybe we should do this check in `clusterIsValidPacket()` instead. You can decide.

Reply:
Remove the call to `getNodeFromLinkAndMsg` for light type and moved it after the light condition